crates/core/src/client/mod.rs

```diff
@@ -462,6 +458,33 @@ impl KakarotClient for KakarotClientImpl<JsonRpcClient<HttpTransport>> {
                     let eth_tx = starknet_tx.to_eth_transaction(self, None, None, None).await?;
                     let from = eth_tx.from;
                     let to = eth_tx.to;
+                    let mut contract_address = to;
+
+                    // If Address is none, contract address from events
+                    if to.is_none() {
+                        // Check if there is any event and if the first event has any data
+                        if let Some(first_event_data) = events.get(0).and_then(|event| event.data.get(0)) {
+                            let evm_address_felt = *first_event_data;
```

Comment:
- why is the first element in event.data the evm address?
- will it stay this way?
- this happens when some does sendRawTx with to = none and tries to deploy an evm smart contract?

Reply:
yeah, but i don't think its necessarily always the first event. an event is emitted on create: https://github.com/jobez/kakarot/blob/main/src/kakarot/accounts/library.cairo#L95

you can see how clement's deploy script parses the events of the receipt and looks for the ones that have the 'from' of the kkrt_address

https://github.com/kkrt-labs/kakarot/blob/5b2b9bf67a2233d71405561bae62734051180132/scripts/utils/kakarot.py#L97-L101